### ISSUE TITLE
Invalid variable in automation body

### ISSUE THREAD

**Issue by grsh50** [apitable/apitable#1773]
TLDR : Is there any way to avoid getting such errors after copying a body from another automation?

I have a large number of datasheets in many spaces and each of them has its own automation but not as a file node, each automation works the same and I would like to easily copy the body to each of them, is there any way to do this or do I have to do everything manually.

maybe there is some syntax that will help me pass the body correctly?

![2024-11-07 101102](https://github.com/user-attachments/assets/2f96854d-2845-455c-bad6-23338c912023)

![error-2024-11-07 101331](https://github.com/user-attachments/assets/00768a98-3150-4602-a997-6f746c42f215)

**Comment by mr-kelly** [comment]
Direct copying of automation configurations between different datasheets or spaces is not currently supported because the variables (field IDs, node IDs) are unique to each datasheet. Simply copying the JSON body will result in "Invalid variable" errors as the IDs do not exist in the new context.

We have recorded this as a feature request for "Automation Templates" or "Portable Configuration" to support easier migration.

---

目前暂不支持在不同数据表或空间之间直接复制自动化配置，因为变量（如字段 ID、节点 ID）在每个数据表中都是唯一的。直接复制 JSON 配置会导致“无效变量”错误，因为这些 ID 在新环境中并不存在。

我们已将其记录为“自动化模板”或“便携式配置”的功能需求，以便支持更轻松的迁移。
